qemu/tests/numa_cpu.py

```diff
@@ -0,0 +1,139 @@
+import logging
+
+from virttest import error_context
+from virttest.utils_misc import NumaInfo
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Assign cpu to numa node with "-numa cpu", check the numa info in monitor
+    and guest os match with the qemu cli
+    """
+
+    def numa_cpu_guest():
+        """
+        The numa cpus in guest os, linux only
+        """
+        error_context.context("Get cpus in guest os", logging.info)
+        numa_info_guest = NumaInfo(session=session)
+        nodes_guest = numa_info_guest.online_nodes
+        numa_cpu_guest = []
+        for node in nodes_guest:
+            numa_cpus = numa_info_guest.online_nodes_cpus[node]
+            numa_cpus = set([int(v) for v in numa_cpus.split()])
+            numa_cpu_guest.append(numa_cpus)
+        return numa_cpu_guest
+
+    def get_expected_numa_cpus():
+        """
+        Get the expected numa cpus for every node according to the qemu cli
+        """
+
+        def _get_boundary(value, max_value, weight):
+            min_boundary = int(value if value else 0) * weight
+            max_boundary = int(value if value else (max_value - 1)) * weight
+            return (min_boundary, max_boundary)
+
+        error_context.context("Get the expected cpus", logging.info)
+        cpus = params.objects("guest_numa_cpus")
+        expected_cpus = []
+        tmp = {}
+        for cpu in cpus:
+            nodeid = params["numa_cpu_nodeid_%s" % cpu]
+            socket = params.get("numa_cpu_socketid_%s" % cpu)
+            die = params.get("numa_cpu_dieid_%s" % cpu)
+            core = params.get("numa_cpu_coreid_%s" % cpu)
+            thread = params.get("numa_cpu_threadid_%s" % cpu)
+            if vm_arch in ('x86_64', 'i386'):
+                socket_min, socket_max = _get_boundary(socket, vcpu_sockets, socket_weight)
+                die_min, die_max = _get_boundary(die, vcpu_dies, die_weight)
+                core_min, core_max = _get_boundary(core, vcpu_cores, core_weight)
+                thread_min, thread_max = _get_boundary(thread, vcpu_threads, thread_weight)
+                cpu_min = socket_min + die_min + core_min + thread_min
+                cpu_max = socket_max + die_max + core_max + thread_max
+                if nodeid in tmp.keys():
+                    tmp[nodeid] += range(cpu_min, cpu_max+1, 1)
+                else:
+                    tmp[nodeid] = range(cpu_min, cpu_max+1, 1)
+            elif vm_arch in ('ppc64le'):
```

Comment:
Should cover `ppc64`( RHEL7 guest)

Reply:
> Should cover `ppc64`( RHEL7 guest)

I remove ppc64 with the latest update since i thought of rhel8 only support ppc64le, i thought it is for both host and guest...   Will add it back, thanks yihuang @PaulYuuu